src/backend/access/heap/heapam_visibility.c

```diff
@@ -1792,3 +1796,40 @@ HeapTupleSatisfiesVisibility(HeapTuple tup, Snapshot snapshot, Buffer buffer)
 
 	return false;				/* keep compiler quiet */
 }
+
+
+bool
+table_tuple_satisfies_visibility(Relation relation, HeapTuple stup, Snapshot snapshot, Buffer buffer)
+{
+	if (table_tuple_satisfies_visibility_hook)
+		return table_tuple_satisfies_visibility_hook(relation, stup, snapshot, buffer);
+
+	return HeapTupleSatisfiesVisibility(stup, snapshot, buffer);
+}
+
+TM_Result
+table_tuple_satisfies_update(Relation relation, HeapTuple stup, CommandId curcid, Buffer buffer)
+{
+	if (table_tuple_satisfies_update_hook)
+		return table_tuple_satisfies_update_hook(relation, stup, curcid, buffer);
+
+	return HeapTupleSatisfiesUpdate(stup, curcid, buffer);
+}
+
+HTSV_Result
+table_tuple_satisfies_vacuum(Relation relation, HeapTuple stup, TransactionId OldestXmin, Buffer buffer)
+{
+	if (table_tuple_satisfies_vacuum_hook)
+		return table_tuple_satisfies_vacuum_hook(relation, stup, OldestXmin, buffer);
+
+	return HeapTupleSatisfiesVacuum(stup, OldestXmin, buffer);
+}
+
+HTSV_Result
+table_tuple_satisfies_vacuum_horizon(Relation relation, HeapTuple htup, Buffer buffer, TransactionId *dead_after)
```

Comment:
what is the best way to make sure future direct references to HeapTupleSatisfiesVacuumHorizon would be detected automatically ?

Reply:
cant think of an efficient way but add a comment to ensure the `table_tuple_satisfies*` API should be called  and add an assert in these functions to make sure table variables do not go to the functions

same thing for 4 of these functions

```221 extern bool HeapTupleSatisfiesVisibility(HeapTuple stup, Snapshot snapshot,
222                                          Buffer buffer);
223 extern TM_Result HeapTupleSatisfiesUpdate(HeapTuple stup, CommandId curcid,
224                                           Buffer buffer);
225 extern HTSV_Result HeapTupleSatisfiesVacuum(HeapTuple stup, TransactionId OldestXmin,
226                                             Buffer buffer);
227 extern HTSV_Result HeapTupleSatisfiesVacuumHorizon(HeapTuple stup, Buffer buffer,
228                                                    TransactionId *dead_after);
```

Reply:
It should work if we add parameter `Relation relation` to functions `table_tuple_satisfies*` and do bashfish hook check inside these functions. Once there is new reference from community in the future, aurora code build will fail and we can fix the community merge accordingly.

Reply:
Thank you. As discussed, I've added a new parameter `Relation relation` inside the `HeapTupleSatisfies*` functions and use the `relation` parameter to detect if it is a Table Variable.